### PR TITLE
[WIP] Make DagRun and Dag pylint-compatible

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by turbaszek** [comment]
I'm afraid that this will require more changes due to cyclic imports :<

File: airflow/models/dag.py [hunk 1]
```diff
@@ -1865,22 +1925,31 @@ def deactivate_deleted_dags(cls, alive_dag_filelocs: List[str], session=None):
...
-                    continue
+            cls._deactivate_deleted_dags_in_models(alive_dag_filelocs, dag_models)
             session.commit()
         except Exception:
             session.rollback()
             raise
 
+    @classmethod
+    def _deactivate_deleted_dags_in_models(cls, alive_dag_filelocs, dag_models):
+        """Loop through all models and deactivate te DAGs."""
```
Comment (mik-laj) [inline airflow/models/dag.py]:
```suggestion
        """Loop through all models and deactivate DAGs."""
```

File: airflow/models/dag.py [hunk 2]
```diff
@@ -1469,7 +1515,7 @@ def create_dagrun(self,
                       run_id=None,
                       start_date=None,
                       external_trigger=False,
-                      conf=None,
+                      conf_dict=None,
```
Comment (kaxil) [inline airflow/models/dag.py]: This is a breaking change, many users use keyword args

Comment (potiuk) [reply airflow/models/dag.py]: True. Will go back with that - besides seems some cyclic imports need to be solved as well. I ill let the other "pylint" changes to be merged and will take a stab (I think one last time :) ) into this one ... We are basically 3 files to go :)

Comment (potiuk) [reply airflow/models/dag.py]: Resolved.

File: airflow/models/dag.py (hunk 1 above)
Comment (potiuk) [reply airflow/models/dag.py]: Resolved.

**Comment by potiuk** [comment]
I want to rebase that one on top of #10499  to see if the cyclic dependencies problem will be fixed,

**Comment by potiuk** [comment]
This is the last one @BasPH ...... https://lists.apache.org/thread.html/f4940d36e98ded96a2473bb2ccdfa4cc648faa2c1334b2aa901c0bba%40%3Cdev.airflow.apache.org%3E

**Comment by potiuk** [comment]
It will likely fail with cyclic impots we will have to resolve though :(

**Comment by potiuk** [comment]
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.models.dag -> airflow.operators.subdag_operator) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.models -> airflow.models.dag -> airflow.sensors.external_task_sensor) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.models -> airflow.models.dagbag -> airflow.models.dag -> airflow.sensors.external_task_sensor -> airflow.operators.dummy_operator) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.api.common.experimental.get_task_instance -> airflow.models -> airflow.models.dag -> airflow.operators.subdag_operator) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.models -> airflow.models.dagbag -> airflow.models.serialized_dag -> airflow.serialization.serialized_objects -> airflow.models.dag -> airflow.sensors.external_task_sensor -> airflow.operators.dummy_operator) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.models -> airflow.models.dagbag -> airflow.models.serialized_dag -> airflow.models.dag -> airflow.sensors.external_task_sensor -> airflow.operators.dummy_operator) (cyclic-import)
airflow/providers/amazon/aws/example_dags/example_emr_job_flow_automatic_steps.py:1:0: R0401: Cyclic import (airflow.api.common.experimental -> airflow.models -> airflow.models.dag -> airflow.operators.subdag_operator -> airflow.api.common.experimental.get_task_instance) (cyclic-import)

**Comment by potiuk** [comment]
Tomorrow :)

**Pull request closed without merging**